packages/layout/src/kit/Types.ts

```diff
@@ -1,52 +1,87 @@
-import { ComponentClassNameProps, NestedClassName, PolymorphicComponentPropsWithRef } from '@contember/utilities'
-import { ElementType, ReactElement, ReactNode } from 'react'
+import { ComponentClassNameProps, NestedClassName, PolymorphicComponent } from '@contember/utilities'
+import { ComponentProps, ReactNode } from 'react'
 import { GetLayoutPanelsStateContextType, OwnPanelProps, PanelState } from '../primitives'
 
-export type OwnFrameProps =
-	& ComponentClassNameProps
-	& {
-		bodyFooter?: ReactNode;
-		bodyHeader?: ReactNode;
-		footer?: ReactNode;
-		footerIsFixed?: boolean;
-		footerClassName?: NestedClassName;
-		header?: ReactNode;
-		headerIsFixed?: boolean;
-		headerClassName?: NestedClassName;
-		minimumFooterHeight?: number;
-		minimumHeaderHeight?: number;
-	}
-
-export type FrameProps<C extends ElementType> = PolymorphicComponentPropsWithRef<C, OwnFrameProps>
-
-export type FrameComponentType =
-	& (<C extends ElementType = 'div'>(props: FrameProps<C>) => React.ReactElement | null)
-	& { displayName?: string | undefined }
-
-export type OwnContentPanelProps =
-	& Omit<ComponentClassNameProps, 'children'>
-	& {
-		basis?: number;
-		body?: ReactNode | ((state: PanelState, panelsState: GetLayoutPanelsStateContextType) => ReactNode);
-		children?: never;
-		footer?: ReactNode | ((state: PanelState, panelsState: GetLayoutPanelsStateContextType) => ReactNode);
-		header?: ReactNode | ((state: PanelState, panelsState: GetLayoutPanelsStateContextType) => ReactNode);
-		maxWidth?: number | false | null | undefined;
-		minWidth?: number | null | undefined;
-		priority?: number | null | undefined;
-	}
-
-export type ContentPanelProps<C extends ElementType> = PolymorphicComponentPropsWithRef<C, OwnContentPanelProps>
-
-export type ContentPanelComponentType =
-	& (<C extends ElementType = 'section'>(props: ContentPanelProps<C>) => ReactElement | null)
-	& {
-		BASIS: number;
-		MAX_WIDTH: number;
-		MIN_WIDTH: number;
-		NAME: string;
-		displayName?: string | undefined;
-	}
+export interface OwnFrameProps extends ComponentClassNameProps {
```

Comment:
please prefer type with intersection over extends, it produces better generated apidoc

Reply:
OK. However, note that `extends` will error on you when types overlap in any of their props that are not matching and therefor are more error prone to future changes.

Reply:
Done